app/src/main/java/com/gst/gusto/review/fragment/InstaReviewFragment.kt

```diff
@@ -0,0 +1,142 @@
+        savedInstanceState: Bundle?
+    ): View? {
+
+        binding = FragmentGalleryReviewBinding.inflate(inflater, container, false)
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initView()
+        setToast()
+        pagingRecyclerview()
+    }
+
+    fun initView(){
+
+        viewModel.instaReviews.observe(viewLifecycleOwner){
+            adapter = InstaReviewAdapter(it, context
+            ) { position ->
+                val bundle = Bundle()
+                bundle.putLong(
+                    "reviewId",
+                    viewModel.instaReviews.value?.get(position)?.reviewId ?: -1
+                )     //리뷰 아이디 넘겨 주면 됨
+                Log.d("reviews", "${viewModel.instaReviews.value}")
+                bundle.putString("page", "review")
+                findNavController().navigate(R.id.action_reviewFragment_to_reviewDetail, bundle)
+            }
+            binding.apply {
+                // 클릭 리스너 부분
+                recyclerView.adapter = adapter
+                var count = 0
+                viewModel.instaReviews.value?.forEach { review ->
+                    review?.let {
+                        Log.d("reviews", "Review ID: ${count} + ${it.reviewId}")
+                        count++
+                    }
+                }
+                val size = resources.getDimensionPixelSize(R.dimen.one_dp)
+                val color = Color.WHITE
+                val itemDecoration = GridItemDecoration(size, color)
+                recyclerView.addItemDecoration(itemDecoration)
+                recyclerView.layoutManager = GridLayoutManager(activity, 3)
+                adapter.addLoading()
+            }
+        }
+    }
+
+
+    fun pagingRecyclerview(){
+        binding.recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+
+                var flag = 0
+
+                val rvPosition = (recyclerView.layoutManager as LinearLayoutManager?)?.findLastCompletelyVisibleItemPosition()
+
+                // 리사이클러뷰 아이템 총개수 (index 접근 이기 때문에 -1)
+                val totalCount = recyclerView.adapter?.itemCount?.minus(1)
+                    // 페이징 처리
+                viewModel.instaPagingdata.observe(viewLifecycleOwner){
+                    if(rvPosition == totalCount && it.second == true && flag == 0) {
+                        getReviewPaging()
+                        flag = 1
+                    }
+                    val handler = Handler(Looper.getMainLooper())
+                    handler.postDelayed({
+                        it.second?.takeIf { !it }?.let {
+                            adapter.removeLoading()
+                        }
+                    }, 3000)
+                }
+            }
+        })
+    }
```

Comment:
호출 구조가 복잡하고 비효율적으로 구성되었네요 !
LiveData 에 대한 observe 처리는 처음 1번만 해주면 되는데, 지금은 스크롤이 끝에 닿을 때마다 다시 새로 해주고 있어요.
instaPagingData 의 second 값 (hasNext) 가 필요해서 이렇게 된 것 같은데, 그럼 그냥 이 작업들을 모두 VIewModel 에 위임하면 깔끔할 것 같아요

```suggestion
    fun pagingRecyclerview(){
        binding.recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                super.onScrolled(recyclerView, dx, dy)

                // 이건 UI 에서만 알 수 있는 정보니까 여기서 구해야해요.
                // 이름도 조금 더 정확하게 바꿔봤어요.
                val lastCompletlyVisibleItemPosition = (recyclerView.layoutManager as LinearLayoutManager?)?.findLastCompletelyVisibleItemPosition()
                
                // 나머지 판단(동작)은 뷰모델에 맡겨버렸어요.
                viewModel.onScrolled(lastCompletelyVisibleItemPosition) 
            }
        })
    }
```